ui/src/conceptList.tsx

```diff
@@ -0,0 +1,103 @@
+import Typography from "@mui/material/Typography";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import React, { useContext, useEffect, useMemo, useState } from "react";
+import { useParams } from "react-router-dom";
+import { EntityInstancesApiContext } from "./apiContext";
+import { State } from "./state";
+
+// Row acts as a bridge between the data format returned from the API and the
+// format expected by DataGrid.
+class Row {
+  id = 0;
+  [x: string]: string | number | boolean | undefined;
+}
+
+type ConceptListProps = {
+  underlay: string;
+  state: State;
+  filter: string;
+};
+
+export default function ConceptList(props: ConceptListProps) {
+  const [error, setError] = useState<Error | null>(null);
+  const [rows, setRows] = useState<Array<Row> | null>(null);
+
+  const params = useParams<{ group: string; criteria: string }>();
+  const api = useContext(EntityInstancesApiContext);
+
+  const columns: GridColDef[] = useMemo(() => {
```

Comment:
opt nit: in ES6, you can omit the function brackets / return and just put an expression, even multiline as in this case

Reply:
Done